### PR TITLE
Release 1.1 - refactor, replace file with db table, discuss improvements

### PR DESCRIPTION
**Pull request by mrow4a**
Replacing https://github.com/owncloud/files_paperhive/pull/12. Supporting 10.0 only. Release TODOs:

- [x] Adjust paperhive to work with oc10
- [x] Adjust icon to standard one
- [x] Refactor code and clean-up
- [x] Replace reading from disk for BookID with db table
- [x] Unit tests finished
- [x] Discuss will ask to open new window instead of blocked window

### CODE REVIEW

**Comment by mrow4a** [comment]
@DeepDiver1975 I think we can merge it - I will later continue to release process. Maybe some additional manual tests from @phil-davis to be sure?

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,18 @@
 language: php
 sudo: false
 php:
-  - 5.6
-  - 7.0
-  - 7.1
+#  - 5.6
```
Comment (DeepDiver1975) [inline .travis.yml]: bring back?

File: .travis.yml [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
+#  - 7.0
+#  - 7.1
   - 7.2
 
 env:
   global:
-    - CORE_BRANCH=master
```
Comment (DeepDiver1975) [inline .travis.yml]: test with master as well

File: .travis.yml [hunk 3]
```diff
@@ -1,26 +1,18 @@
...
+    - CORE_BRANCH=stable10
     - APP=files_paperhive
   matrix:
     - DB=sqlite;
 
-matrix:
-  include:
-    - php: 5.4
-      env: DB=sqlite CORE_BRANCH=stable9
```
Comment (DeepDiver1975) [inline .travis.yml]: older version are dead?

File: appinfo/Migrations/Version20181006201617.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace OCA\Files_PaperHive\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use OCP\Migration\ISchemaMigration;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
```
Comment (DeepDiver1975) [inline appinfo/Migrations/Version20181006201617.php]: remove generated comment

File: appinfo/Migrations/Version20181006201617.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<?php
```
Comment (DeepDiver1975) [inline appinfo/Migrations/Version20181006201617.php]: license header missing

File: appinfo/routes.php [hunk 6]
```diff
@@ -19,25 +19,32 @@
...
+$app->registerRoutes(
+	$this,
 	[
-		'name' => 'PaperHive#getPaperHiveDocument',
-		'url' => '/ajax/getpaperhivedocument',
-		'verb' => 'GET'
+		'routes' => [
+			[
+				'name' => 'PaperHive#getPaperHiveDetails',
+				'url' => '/getpaperhivedetails',
```
Comment (DeepDiver1975) [inline appinfo/routes.php]: remove get from url - get is already in the method/verb

File: .travis.yml (hunk 2 above)
Comment (mrow4a) [reply .travis.yml]: I support only 10.0-10.99. Did not allow 10.1 yet.

File: .travis.yml (hunk 3 above)
Comment (mrow4a) [reply .travis.yml]: yes, only 10.0.X

**Comment by mrow4a** [comment]
@DeepDiver1975 is there anything else I should address?

**Pull request merged**